pkgs/development/python-modules/hypothesis/default.nix

```
@@ -4,59 +4,69 @@
 , fetchFromGitHub
 , attrs
 , pexpect
-, doCheck ? true
 , pytestCheckHook
 , pytest-xdist
 , sortedcontainers
 , pythonOlder
 }:
 
-buildPythonPackage rec {
-  pname = "hypothesis";
-  version = "6.46.10";
-  format = "setuptools";
+let
+  self = buildPythonPackage rec {
+    pname = "hypothesis";
+    version = "6.46.10";
+    format = "setuptools";
 
-  disabled = pythonOlder "3.7";
+    disabled = pythonOlder "3.7";
 
-  src = fetchFromGitHub {
-    owner = "HypothesisWorks";
-    repo = "hypothesis";
-    rev = "hypothesis-python-${version}";
-    hash = "sha256-eQ7Ns0k1hOVw8/xiINMei6GbQqDHXrBl+1v8YQeFO9Q=";
-  };
+    src = fetchFromGitHub {
+      owner = "HypothesisWorks";
+      repo = "hypothesis";
+      rev = "hypothesis-python-${version}";
+      hash = "sha256-eQ7Ns0k1hOVw8/xiINMei6GbQqDHXrBl+1v8YQeFO9Q=";
+    };
+
+    sourceRoot = "${src.name}/hypothesis-python";
 
-  postUnpack = "sourceRoot=$sourceRoot/hypothesis-python";
+    propagatedBuildInputs = [
+      attrs
+      sortedcontainers
+    ];
 
-  propagatedBuildInputs = [
-    attrs
-    sortedcontainers
-  ];
+    checkInputs = [
+      pexpect
+      pytest-xdist
+      pytestCheckHook
+    ];
 
-  checkInputs = [
-    pexpect
-    pytest-xdist
-    pytestCheckHook
-  ];
+    # Avoid cyclic dependency with pytest
+    doCheck = false;
 
-  inherit doCheck;
+    pytestFlagsArray = [
+      "tests/cover"
+      "-p no:cacheprovider" # tries to create cache dir in invalid location
+      "-n $NIX_BUILD_CORES"
+    ];
 
-  # This file changes how pytest runs and breaks it
-  preCheck = ''
-    rm tox.ini
-  '';
+    pythonImportsCheck = [
+      "hypothesis"
+    ];
 
-  pytestFlagsArray = [
-    "tests/cover"
-  ];
+    # avoid coverage, and allow for xdist to be used
+    preCheck = ''
+      rm tox.ini
+    '';
 
-  pythonImportsCheck = [
-    "hypothesis"
-  ];
+    passthru.tests.hypothesisWithTests = self.overrideAttrs (_: {
```

```suggestion
    passthru.tests.hypothesisWithTests = hypothesis.overrideAttrs (_: {
```

That should allow getting rid of the whole `let self = ...; in self` construct.

---

It does mean when overriding you also need to pass in the overridden hypothesis.

---

Hm, perhaps you could make use of `finalAttrs` instead of overrides then.